uber/models.py

```diff
@@ -1062,6 +1193,23 @@ class Attendee(MagModel, TakesPaymentMixin):
     group_id = Column(UUID, ForeignKey('group.id', ondelete='SET NULL'), nullable=True)
     group = relationship(Group, backref='attendees', foreign_keys=group_id, cascade='save-update,merge,refresh-expire,expunge')
 
+    # NOTE: The cascade relationships for promo_code do NOT include
+    # "save-update". During the preregistration workflow, before an Attendee
+    # has paid, we create ephemeral Attendee objects that are saved in the
+    # cherrypy session, but are NOT saved in the database. If the cascade
+    # relationships specified "save-update" then the Attendee would
+    # automatically be inserted in the database when the promo_code is set on
+    # the Attendee object (which we do not want until the attendee pays).
+    #
+    # The practical result of this is that we must manually set promo_code_id
```

Comment:
I think that's very much worth it for a bunch of reasons, and this is another one.